samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp

```diff
@@ -793,7 +794,7 @@
 
  <!-- save & publish -->
   <h:commandButton  value="#{assessmentSettingsMessages.button_unique_save_and_publish}" type="submit" styleClass="active" rendered="#{assessmentSettings.hasQuestions}"
-      action="#{assessmentSettings.getOutcomePublish}" onclick="setBlockDivs();updateItemNavigation(false);" >
+      action="#{assessmentSettings.getOutcomePublish}" onclick="SPNR.disableControlsAndSpin(this, null);setBlockDivs();updateItemNavigation(false);" >
```

Comment:
All these changes look good on the surface, but we need to keep in mind the following:

If the element's onclick, or the form itself, has other JavaScript that runs (for example to do client side validation and abort form submission if there are errors), then applying the spinner before that JS runs could lead to the page being refreshed with the intention of allowing the user to modify for values, but the spinner would still be applied and all form controls would be disallowed. This would force the user to refresh the page, possibly losing all their inputted data.

So my question is: do these forms or onclicks have further client side validation? If so, the `SPNR.disableControlsAndSpin()`  should be applied within those routines as appropriate, not before they execute.

Reply:
@bjones86 I haven't seen any client side validation, but I have just changed calls order. Thanks!